lib/fetch_book_data.rb

```diff
@@ -0,0 +1,53 @@
+require 'net/http'
+
+# Service object for pulling in book data from the CMS
+class FetchBookData
+  CMS_API_URL = Rails.application.secrets.cms_api_url
+  SUBJECTS_URL = "#{CMS_API_URL}snippets/subjects/?format=json"
+  TITLES_URL = "#{CMS_API_URL}v2/pages/?type=books.Book&format=json&limit=250&fields=title,book_subjects,book_state"
+  TIMEOUT = 1
+
+  def subjects
+    @subjects ||= fetch_subjects
+  end
+
+  def titles
+    @titles ||= fetch_titles
+  end
+
+  def fetch_subjects
+    results = cms_fetch(SUBJECTS_URL)
+    return [] if results.blank?
+
+    results.collect { |subject| subject.fetch('name', nil) }
+  end
+
+  def fetch_titles
+    results = cms_fetch(TITLES_URL)
+    return [] if results.blank?
+
+    items = results.fetch('items', [])
+    items.collect{ |book| book.fetch('title', nil) }
+  end
+
+  private ###################
+
+  def cms_fetch(uri)
+    uri = URI(uri)
+
+    begin
+      Net::HTTP.start(uri.host, uri.port, use_ssl: true, read_timeout: TIMEOUT) do |http|
+        response = Net::HTTP.get_response(uri)
+        body = JSON.parse(response.body)
+        return body
+      end
+    rescue Net::ReadTimeout => ee
+      Raven.capture_message("Fetching book data from the CMS timed out")
+      return nil
```

Comment:
suggestion.  this does work (of course, only in that you're capturing the exception.  i would still consider sending the ee.message and stack trace to Raven tho for ease of debugging).   A better pattern to follow, over returning nil is to return an error object that has contextual information.  This is a much better "code reuse" pattern to follow and more OO.  might be over-kill just for this specific situation, but mentioned it because it's a good practice to follow with API calls like this. 

more info... "nil values are the worst" by Avdi Grimm https://www.rubytapas.com/2017/01/31/two-screencasts-two-ways-eradicate-ruby-nil-values/

Reply:
> sending the ee.message and stack trace to Raven

Totally agree, that's a great idea when the exception is something other than a timeout (gonna send the `extra` option to Raven) but do we really need it when the exception is just a timeout?

Reply:
Oh, we already send to Raven the whole exception when the exception is something other than a timeout. Ok, still gonna send it when it's a timeout 😄 .